ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+    valid[:]=False
+    for nv in range(0,nvars):
+        varname = varlist[nv,0]
+        if varname in ncvarlist:
+            valid[nv]=True
+        print(varlist[nv,0]+'   '+str(valid[nv]))
+
+    #check for vector variables in file. 
+    #If at least one component of the vector is not in input file, 
+    #then the entire vector pair will be ignorned
+    validv=np.zeros(shape=(nvpairs),dtype=bool)
+    validv[:]=False
+    for nv in range(0,nvpairs):
+        vecnames = veclist[nv,0,:]
+        for nv2 in range(0,2):
+            vecname=vecnames[nv2]
+            if vecname in ncvarlist:
+                validv[nv]=True
+            else:
+                validv[nv]=False
+                break
+        print(veclist[nv,0,0]+' and '+veclist[nv,0,1]+'  '+str(validv[nv]))
+
+    #----------------------------------------------------------------------
+    # loop over the output resolutions 
+    #----------------------------------------------------------------------
+
+    for jj in range(0,len(dstgrds)):
+
+        outres = dstgrds[jj]+'x'+dstgrds[jj]
+        outgrid = dstgrds[jj]
+
+        if outgrid == '0p25':
+            lond = np.arange(0, 360.0, 0.25)
+            latd = np.arange(-90.0, 90.25, 0.25)
+        if outgrid == '0p5':
+            lond = np.arange(0, 360.0, 0.5)
+            latd = np.arange(-90.0, 90.5, 0.5)
+        if outgrid == '1p0':
+            lond = np.arange(0, 360.0, 1.0)
+            latd = np.arange(-90.0, 90.1, 1.0)
+
+        wgtsfile1 = nemsrc+'/'+'tripole.mx025.Ct.to.'+dsttype[0]+dstgrds[jj]+'.bilinear.nc'        
+        wgtsfile2 = nemsrc+'/'+'tripole.mx025.Ct.to.'+dsttype[0]+dstgrds[jj]+'.conserve.nc'
+        if model == 'MOM':
+            wgtsfile3 = nemsrc+'/'+'tripole.mx025.Cu.to.Ct.bilinear.nc'    
+            wgtsfile4 = nemsrc+'/'+'tripole.mx025.Cv.to.Ct.bilinear.nc'
+        if model == 'CICE':
+            wgtsfile3 = nemsrc+'/'+'tripole.mx025.Bu.to.Ct.bilinear.nc'    
+            wgtsfile4 = nemsrc+'/'+'tripole.mx025.Bu.to.Ct.bilinear.nc'
+
+        print(wgtsfile1)
+        print(wgtsfile2)
+        print(wgtsfile3)
+        print(wgtsfile4)
+
+        #Extract regridding weights from netcdf files
+        rgrdf1=nc.Dataset(wgtsfile1)
+        S1=rgrdf1['S'][:].copy()
+        row1=rgrdf1['row'][:]
+        col1=rgrdf1['col'][:]
+        frac_b1=rgrdf1['frac_b'][:]
+        n_s1=len(S1)
+
+        rgrdf2=nc.Dataset(wgtsfile2)
+        S2=rgrdf2['S'][:].copy()
+        row2=rgrdf2['row'][:]
+        col2=rgrdf2['col'][:]
+        frac_b2=rgrdf2['frac_b'][:]
+        n_s2=len(S2)
+
+        rgrdf3=nc.Dataset(wgtsfile3)
+        S3=rgrdf3['S'][:].copy()
+        row3=rgrdf3['row'][:]
+        col3=rgrdf3['col'][:]
+        frac_b3=rgrdf3['frac_b'][:]
+        n_s3=len(S3)
+
+        rgrdf4=nc.Dataset(wgtsfile4)
+        S4=rgrdf4['S'][:].copy()
+        row4=rgrdf4['row'][:]
+        col4=rgrdf4['col'][:]
+        frac_b4=rgrdf4['frac_b'][:]
+        n_s4=len(S4)
```

Comment:
`netCDF4` affords lazy-loading of the dataset.  Why are we extracting that data upfront?  This doesn't seem like a good use of the library.